### ISSUE TITLE
An ability to customize background color under keyboard?

### ISSUE THREAD

**Issue by kirillzyusko** [mateusz1913/react-native-avoid-softinput#48]
### Feature request

Hi, @mateusz1913 

First of all I wanted to say thank you for this library!

I'm a little bit curious whether it's possible to customize background color of the view, which is shown under keyboard. Is it possible to make it transparent, for example?

I'm asking, because on Android this space has a white color and it's not moving along with the keyboard (it's moving a little bit slower). And as a result we see how white space may be shown over the content.

If it's not clear, what I'm describing - let me know and I can capture video/screenshots and explain it better :)

**Comment by mateusz1913** [comment]
Hi @kirillzyusko 

It's possible to just set different background color of app's root view, you can do it:
a) statically:
on Android, you can specify theme values (not sure but maybe android:windowBackground), or try to change root view background color inside your MainActivity
on iOS, you can change background color of RCTRootView inside your project's `AppDelegate` (there is quite old, but still relevant [article](https://medium.com/@Krizzu/how-to-change-background-color-of-rootview-in-react-native-3279417c34bb))
b) dynamically:
you can use one of 3rd party libs to do it from JS code (e.g. https://github.com/kev-chen/react-native-root-view-background-color)

I would not want to include that logic inside this library, as I think it's a little out of scope and there are already existing solutions to achieve it, so implementing it in this lib would be redundant

I am closing issue for now, but let me know if provided solutions work for you (if not, please reopen with details, why these are not working in your case 👋  and I will try to investigate)

**Comment by kirillzyusko** [comment]
Hi @mateusz1913 

Thank you for your input. I think I had to describe the problem more detailed 😅 

If I'm using `react-native-root-viw-background-color` or any other library that changes root view background color via `view.getRootView().setBackgroundColor` method - it doesn't work.

I've tried to add the library to your example project. I'm using it in `FormExample.tsx` and use library in the following way:

```diff
const onFocusEffect = useCallback(() => {
    AvoidSoftInput.setAdjustNothing();
    AvoidSoftInput.setEnabled(true);
+   RootViewBackgroundColor.setBackground(0, 255, 0, 1);
    return () => {
      AvoidSoftInput.setEnabled(false);
      AvoidSoftInput.setDefaultAppSoftInputMode();
    };
  }, []);
```

I can see green background during a navigation transition, but when I touch `TextInput` field the color of view under keyboard is gray. Here is a video:

https://user-images.githubusercontent.com/22820318/160285814-faf0fd8d-1e00-4880-a809-232dbc45ba8c.mov

And screenshot breakdown:

<table>
<thead>
<td>Transition</td>
<td>Keyboard disappearing</td>
</thead>
<tr>
<td>
<img width="481" alt="image" src="https://user-images.githubusercontent.com/22820318/160285897-dfae7ac3-38eb-4ce9-b6c3-055db388837e.png">
</td>
<td>
<img width="481" alt="image" src="https://user-images.githubusercontent.com/22820318/160286131-d6a6d9ca-63ee-4b6b-9d4f-0dce452dbe66.png">
</td>
</tr>
<tr>
<td>I see green background, so the library works</td>
<td>Here when keyboard is hidden I still see gray view, and I want to make this view transparent - is it possible?</td>
</tr>
</table>

Am I missing something important?

**Comment by mateusz1913** [comment]
Oh, I see. Form example is using ScrollView, so you would have try playing with ScrollView backgroundColor property in style prop. If it won't work, then you can try changing hideAnimationDelay to 0

**Comment by kirillzyusko** [comment]
@mateusz1913 I've tried to specify `backgroundColor` for `ScrollView` as "transparent", "#FFFFFF00" and "rgba(255, 255, 255, 0)" - still have the same problem (view under background doesn't have transparent color)

Wrapping `ScrollView` in additional `AvoidSoftInputView` with `hideAnimationDelay={0}` didn't help either.

When you add this view - you are traversing the tree to nearest `ScrollView` and apply the same style for the view under keyboard?

Any ideas what else I can try to see the real content beneath keyboard straight away after its disappearing?

**Comment by mateusz1913** [comment]
Form example uses `AvoidSoftInput` module, so you can try to use `AvoidSoftInput.setHideAnimationDelay(0)`

**Comment by mateusz1913** [comment]
Oh sorry @kirillzyusko , when I meant hideAnimationDelay, I actually thought about hideAnimationDuration :D , Default value is 220ms, so in your case, you can use `AvoidSoftInput.setHideAnimationDuration(0)`

**Comment by kirillzyusko** [comment]
> Form example uses AvoidSoftInput module, so you can try to use AvoidSoftInput.setHideAnimationDuration(0)

Nice 👍. I was confused a little bit, because as I can understand from documentation, if I'm using methods from module directly, then I don't need to use `AvoidSoftInputView`.

> Oh sorry @kirillzyusko , when I meant hideAnimationDelay, I actually thought about hideAnimationDuration :D , Default value is 220ms, so in your case, you can use AvoidSoftInput.setHideAnimationDuration(0)

I've tried to set it. It looks better but still feels slightly not native. I've recorder it and then slowed it down:

https://user-images.githubusercontent.com/22820318/160290016-43864bbb-52c1-48e0-9b99-65a09e4b2fe4.mov

Yes, the view disappears instantly, but still it disappears only after keyboard is getting hidden. Is it a problem of Android API and missing support for `keyboardWillHide` event?

**Comment by mateusz1913** [comment]
There is no additional view disappearing - it's applied padding that is removed from ScrollView. I am afraid that apart from setting animation duration to 0 or changing background color, there is nothing more that can be done here
